api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -308,12 +309,12 @@ func (mw *azureManagedMachinePoolWebhook) ValidateDelete(ctx context.Context, ob
 		return nil, nil
 	}
 
-	return nil, errors.Wrapf(validateLastSystemNodePool(mw.Client, m.Labels, m.Namespace), "if the delete is triggered via owner MachinePool please refer to trouble shooting section in https://capz.sigs.k8s.io/topics/managedcluster.html")
+	return nil, errors.Wrapf(validateLastSystemNodePool(mw.Client, m.Labels, m.Namespace, m.Annotations), "if the delete is triggered via owner MachinePool please refer to trouble shooting section in https://capz.sigs.k8s.io/topics/managedcluster.html")
 }
 
 // validateLastSystemNodePool is used to check if the existing system node pool is the last system node pool.
 // If it is a last system node pool it cannot be deleted or mutated to user node pool as AKS expects min 1 system node pool.
-func validateLastSystemNodePool(cli client.Client, labels map[string]string, namespace string) error {
+func validateLastSystemNodePool(cli client.Client, labels map[string]string, namespace string, nodePoolAnnotations map[string]string) error {
```

Comment:
```suggestion
func validateLastSystemNodePool(cli client.Client, labels map[string]string, namespace string, annotations map[string]string) error {
```

This feels slightly cleaner since we're just calling labels "labels" and namespace "namespace"

Reply:
Sounds good.